aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -347,4 +347,36 @@ static inline void checkUplo(const std::string& uplo) {
     "Expected UPLO argument to be 'L' or 'U', but got ", uplo);
 }
 
+static inline void checkSameDevice(const std::string& fn_name, Tensor result, Tensor input, const std::string& result_name = "result") {
+  TORCH_CHECK(
+      result.device() == input.device(),
+      fn_name,
+      ": Expected ", result_name, " and input tensors to be on the same device, but got ",
+      result_name, " on ", result.device(), " and input on ", input.device());
+}
+
+// Check the dtype of result and input tensors (for _out variants).
+// Most linear algebra functions have the same dtype for input and output
+// (either floating or complex type input), so we can check whether both result and input are compatible.
+// According to https://github.com/pytorch/pytorch/wiki/Developer-FAQ#how-does-out-work-in-pytorch
+// if input is complex result can only be complex, if input is float result can be both float and complex
```

Comment:
This is true but a little misleading since we allow other casts, too. This comment should probably just refer to the use of the canCast() function.